src/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -2196,23 +2194,26 @@ public int GetLowerBound(int dimension)
 #if REAL_MULTIDIM_ARRAYS
             if (!IsSzArray)
             {
-                if ((dimension >= Rank) || (dimension < 0))
+                int rank = Rank;
+                if (dimension != 0 && (uint)dimension >= rank)
```

Comment:
I do not think comparing dimension with `0` first helps here - since the rank has been computed already.